packages/server/graphql/types/User.ts

```diff
@@ -442,19 +442,55 @@ const User: GraphQLObjectType<any, GQLContext> = new GraphQLObjectType<any, GQLC
       },
       resolve: async ({id: userId}, {reflectionId}, {dataLoader}) => {
         const retroReflection = await dataLoader.get('retroReflections').load(reflectionId)
-        if (!retroReflection) return []
+        if (!retroReflection) {
+          return standardError(new Error('Invalid reflection id'), {userId})
+        }
         const {meetingId} = retroReflection
-        const meetingMemberId = toTeamMemberId(meetingId, userId)
-        const [viewerMeetingMember, reflectionGroups] = await Promise.all([
+        const meetingMemberId = MeetingMemberId.join({meetingId, userId})
+        const r = await getRethink()
+        const [viewerMeetingMember, reflections] = await Promise.all([
           dataLoader.get('meetingMembers').load(meetingMemberId),
-          dataLoader.get('retroReflectionGroupsByMeetingId').load(meetingId)
+          r
+            .table('RetroReflection')
+            .getAll(meetingId, {index: 'meetingId'})
+            .filter({isActive: true})
+            .orderBy('createdAt')
+            .run()
         ])
-        const {meetingId: viewerMeetingId} = viewerMeetingMember
-        if (viewerMeetingId !== meetingId) {
-          standardError(new Error('Not on team'), {userId})
-          return []
+        if (!viewerMeetingMember) {
+          return standardError(new Error('Not on team'), {userId})
+        }
+        const reflectionsCount = reflections.length
+        const spotlightResultGroupIds = new Set<string>()
+        const spotlightResultGroupSize = Math.min(reflectionsCount, MAX_RESULT_GROUP_SIZE)
+        let currentThresh: number | null = AUTO_GROUPING_THRESHOLD
+        while (currentThresh) {
+          const {groupedReflectionsRes, nextThresh} = groupReflections(reflections, {
+            groupingThreshold: currentThresh,
+            maxGroupSize: reflectionsCount,
+            maxReductionPercent: MAX_REDUCTION_PERCENTAGE
+          })
+          const spotlightGroup = groupedReflectionsRes.find(
+            (group) => group.reflectionId === reflectionId
+          )
+          if (!spotlightGroup) break
+          for (const groupedReflection of groupedReflectionsRes) {
+            const {reflectionGroupId, oldReflectionGroupId} = groupedReflection
+            if (
+              reflectionGroupId === spotlightGroup.reflectionGroupId &&
+              oldReflectionGroupId !== spotlightGroup.oldReflectionGroupId
+            ) {
+              spotlightResultGroupIds.add(oldReflectionGroupId)
+            }
+            currentThresh = nextThresh
+            if (spotlightResultGroupIds.size >= spotlightResultGroupSize) break
```

Comment:
don't we want to break before the maximum is reached?
- save the results as R1
- compute next results R2
- if size R2 > max, return R1
- else R1 = R2